src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt

```diff
@@ -157,42 +163,71 @@ class AccountControllerTest @Autowired constructor(
                 "https://github.com"
             )
 
-            mockMvc.post("/accounts/new") {
-                contentType = MediaType.APPLICATION_JSON
-                content = objectMapper.writeValueAsString(
-                    mapOf(
-                        "name" to noWebsite.name,
-                        "email" to noWebsite.email,
-                        "password" to noWebsite.password,
-                        "bio" to noWebsite.bio,
-                        "birthDate" to noWebsite.birthDate,
-                        "photoPath" to noWebsite.photoPath,
-                        "linkedin" to noWebsite.linkedin,
-                        "github" to noWebsite.github
-                    )
-                )
+            val accountPart = MockPart("account", noWebsite.toJson().toByteArray())
+            accountPart.headers.contentType = MediaType.APPLICATION_JSON
+
+            mockMvc.multipart("/accounts/new") {
+                part(accountPart)
             }.andExpect {
                 status { isOk() }
                 content { contentType(MediaType.APPLICATION_JSON) }
                 jsonPath("$.name") { value(noWebsite.name) }
                 jsonPath("$.email") { value(noWebsite.email) }
                 jsonPath("$.bio") { value(noWebsite.bio) }
                 jsonPath("$.birthDate") { value(noWebsite.birthDate.toJson()) }
-                jsonPath("$.photoPath") { value(noWebsite.photoPath) }
                 jsonPath("$.linkedin") { value(noWebsite.linkedin) }
                 jsonPath("$.github") { value(noWebsite.github) }
                 jsonPath("$.websites.length()") { value(0) }
             }
         }
 
+        @Test
+        fun `should create the account with valid image`() {
+            val accountPart = MockPart("account", testAccount.toJson().toByteArray())
+            accountPart.headers.contentType = MediaType.APPLICATION_JSON
+
+            val file = MockMultipartFile(
+                "photo",
+                "photo.jpeg",
+                MediaType.IMAGE_JPEG_VALUE,
+                "content".toByteArray()
+            )
+
+            val uuid: UUID = UUID.randomUUID()
+            Mockito.mockStatic(UUID::class.java)
+            Mockito.`when`(UUID.randomUUID()).thenReturn(uuid)
```

Comment:
Do you know why the `s are used in kotlin? So odd

Reply:
I found this: https://kotlinlang.org/docs/java-interop.html#escaping-for-java-identifiers-that-are-keywords-in-kotlin

Basicly it serves to escape java methods, variables, etc that are keywords in kotlin. It also serves to escape function names(?) like the tests

Reply:
Interesting